### PR TITLE
Removed the Go path link from the ./eventing/event-delivery.md

### LINKED ISSUE #3309: Remove links to Go code
We have some pages that link to Go code (especially Go types).
Most of the time while reading the documentation is not expected to see that a link brings you to a low-level representation of the resource you're learning.
In addition to that, users might not be familiar with Go and they should not think that Go is required for learning or using Knative.

## Expected Behavior

No link to Go code.

## Additional information

We should find all links to Go code, and then for each of them, we can replace with an example using YAML or the `kn` CLI

_If you want to tackle a specific page or link, please comment on this issue, so that there is no duplicate effort from multiple people_

/good-first-issue

### PR DESCRIPTION
**Pull request by Yashks1994**
Fixes #3309 - Remove links to Go code

## Proposed Changes 

- Removed the Go path link for Delivery Spec
- Removed the Go path link for Destination

### CODE REVIEW

File: docs/eventing/event-delivery.md [hunk 1]
```diff
@@ -72,14 +72,14 @@ after the specified number of retries.
 
 ## Common Delivery Parameters
 
-The `delivery` value must be a [Delivery Spec](https://pkg.go.dev/knative.dev/eventing/pkg/apis/duck/v1?tab=doc#DeliverySpec)
+The `delivery` value must be a Delivery Spec.
```
Comment (pierDipi) [inline docs/eventing/event-delivery.md:75]:
```suggestion
The `delivery` value must be a Delivery Spec, which is a partial schema that is embedded in resources like `Broker`, `Trigger` and `Subscription`.

```yaml
# DeadLetterSink is the sink receiving event that could not be sent to
# a destination.
deadLetterSink:
  ref:
    apiVersion: v1
    kind: Service
    name: my-service
  uri: /my-path

# Retry is the minimum number of retries the sender should attempt when
# sending an event before moving it to the dead letter sink.
retry: 5

// BackoffPolicy is the retry backoff policy (linear, exponential).
backoffPolicy: exponential

# BackoffDelay is the delay before retrying.
# More information on Duration format:
#  - https://www.iso.org/iso-8601-date-and-time-format.html
#  - https://en.wikipedia.org/wiki/ISO_8601
#
# For linear policy, backoff delay is backoffDelay*<numberOfRetries>.
# For exponential policy, backoff delay is backoffDelay*2^<numberOfRetries>.
backoffDelay: PT2S```
```

File: docs/eventing/event-delivery.md [hunk 2]
```diff
@@ -72,14 +72,14 @@ after the specified number of retries.
...
 
 ### deadLetterSink
 
 When present, events that failed to be consumed are sent to the `deadLetterSink`.
 In case of failure, the event is dropped and an error is logged into the system.
 
-The `deadLetterSink` value must be a [Destination](https://pkg.go.dev/knative.dev/pkg/apis/duck/v1#Destination).
+The `deadLetterSink` value must be a Destination.
```
Comment (pierDipi) [inline docs/eventing/event-delivery.md:82]: Can you update the example below (line 87) by replacing `<Destination>` with the below YAML?
```YAML
# DeadLetterSink is the sink receiving event that could not be sent to
# a destination.
deadLetterSink:
  ref:
    apiVersion: v1
    kind: Service
    name: my-service
  uri: /my-path
```

File: docs/eventing/event-delivery.md (hunk 1 above)
Comment (Yashks1994) [reply docs/eventing/event-delivery.md]: Done

File: docs/eventing/event-delivery.md (hunk 2 above)
Comment (Yashks1994) [reply docs/eventing/event-delivery.md:109]: Removed the existing yaml with the new one.

**Review by pierDipi — APPROVED** [review]
/lgtm

**Pull request merged**